internal/confmapprovider/discovery/discoverer.go

```diff
@@ -117,6 +119,38 @@ func newDiscoverer(logger *zap.Logger) (*discoverer, error) {
 	return d, nil
 }
 
+func (d *discoverer) discoveryAuthConfFromEnv(discoveryReceiverRaw map[string]any) (*confmap.Conf, error) {
```

Comment:
nit, since this is a general config resolver pass I don't think we need to focus on the auth application. Maybe something like this?

```suggestion
func (d *discoverer) resolveConfig(discoveryReceiverRaw map[string]any) (*confmap.Conf, error) {
```

Reply:
I have read the CLA Document and I hereby sign the CLA